src/sources/socket/unix.rs

```diff
@@ -66,60 +74,91 @@ impl UnixConfig {
             host_key: None,
             framing: None,
             decoding: default_decoding(),
+            log_namespace: None,
         }
     }
+
+    pub const fn decoding(&self) -> &DeserializerConfig {
+        &self.decoding
+    }
+
+    pub const fn host_key(&self) -> &Option<String> {
+        &self.host_key
+    }
 }
 
 /// Function to pass to `build_unix_*_source`, specific to the basic unix source
 /// Takes a single line of a received message and handles an `Event` object.
-fn handle_events(events: &mut [Event], host_key: &str, received_from: Option<Bytes>) {
+fn handle_events(
+    events: &mut [Event],
+    config: &UnixConfig,
```

Comment:
It looks like all you need from this is the `host_key` , seems like you could just take that as an argument to this function in order to avoid cloning the whole config in the callers.

Reply:
yep, I think that's lying around from a previous version of the changes.